src/Appwrite/Platform/Modules/Proxy/Http/Rules/Function/Create.php

```diff
@@ -84,91 +84,69 @@ public function action(string $domain, string $resourceType, string $resourceId,
         }
 
-        // TODO: @christyjacob remove once we migrate the rules in 1.7.x
-        $ruleId = System::getEnv('_APP_RULES_FORMAT') === 'md5' ? md5($domain->get()) : ID::unique();
-
-        try {
-            $rule = new Document([
-                '$id' => $ruleId,
-                'projectId' => $project->getId(),
-                'projectInternalId' => $project->getInternalId(),
-                'domain' => $domain->get(),
-                'resourceType' => $resourceType,
-                'resourceId' => $resourceId,
-                'resourceInternalId' => $resourceInternalId,
-                'certificateId' => '',
-            ]);
-        } catch (\Throwable $e) {
-            if ($e->getCode() === Exception::DOCUMENT_ALREADY_EXISTS) {
-                throw new Exception(Exception::RULE_ALREADY_EXISTS);
+        // Apex domain prevention due to CNAME limitations
+        if (empty(App::getEnv('_APP_DOMAINS_NAMESERVERS', ''))) {
+            if ($domain->get() === $domain->getRegisterable()) {
+                throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'The instance does not allow root-level (apex) domains.');
             }
+        }
 
-            throw new Exception(Exception::GENERAL_SERVER_ERROR, 'An unexpected error occurred: ' . $e->getMessage());
+        $function = $dbForProject->getDocument('functions', $functionId);
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::RULE_RESOURCE_NOT_FOUND);
         }
 
-        $status = 'created';
+        // TODO: @christyjacob remove once we migrate the rules in 1.7.x
+        $ruleId = System::getEnv('_APP_RULES_FORMAT') === 'md5' ? md5($domain->get()) : ID::unique();
 
+        $status = 'created';
         if (\str_ends_with($domain->get(), $functionsDomain) || \str_ends_with($domain->get(), $sitesDomain)) {
             $status = 'verified';
         }
-
         if ($status === 'created') {
             $target = new Domain(System::getEnv('_APP_DOMAIN_TARGET', ''));
-            $validator = new CNAME($target->get()); // Verify Domain with DNS records
-
+            $validator = new CNAME($target->get());
             if ($validator->isValid($domain->get())) {
                 $status = 'verifying';
-
-                $queueForCertificates
-                    ->setDomain(new Document([
-                        'domain' => $rule->getAttribute('domain')
-                    ]))
-                    ->trigger();
             }
         }
 
-        $rule->setAttribute('status', $status);
-        $rule = $dbForPlatform->createDocument('rules', $rule);
+        $rule = new Document([
+            '$id' => $ruleId,
+            'projectId' => $project->getId(),
+            'projectInternalId' => $project->getInternalId(),
+            'domain' => $domain->get(),
+            'status' => $status,
+            'type' => 'deployment',
```

Comment:
Got confused. Does it need to be `deployment` or `trigger` for functions?

Reply:
Rule's type can only be: api, deployment, redirect